src/algorithms/tracking/ActsGeometryProvider.cc

```diff
@@ -1,6 +1,14 @@
 // SPDX-License-Identifier: LGPL-3.0-or-later
 // Copyright (C) 2022 - 2024 Whitney Armstrong, Wouter Deconinck, Dmitry Romanov
 
+// FIXME: gcc-12 affected by https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105562
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunknown-pragmas"
+#pragma clang diagnostic ignored "-Wunknown-warning-option"
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#include <functional> // IWYU pragma: keep
+#pragma GCC diagnostic pop
+
```

Comment:
Just `-Wno-error=unknown-pragmas ...` until we fix this.

Reply:
What's your suggestion? We already pass `-Wno-unknown-pragmas`, if that's what you are implying.

Reply:
The context of my comment is:
```
// FIXME: gcc-12 affected by https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105562
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wunknown-pragmas"
#pragma clang diagnostic ignored "-Wunknown-warning-option"
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
#include <functional> // IWYU pragma: keep
#pragma GCC diagnostic pop
```
We could address it with
`-Wno-error=unknown-pragmas -Wno-error=unknown-warning-option -Wno-error=maybe-uninitialized`

Reply:
I don't think so. gcc doesn't accept `unknown-warning-option`

Frankly this is a mess on the clang side:
- they parse the GCC pragmas
- there is no way to turn that off
- they fail for some gcc options they don't recognize
- the way to ignore unknown gcc options is with an option that gcc fails on

Reply:
So, I repeat: what's your suggestion and does it work with gcc and clang both?